store/sqlstore/retention_policy_store.go

```diff
@@ -329,118 +335,178 @@ func (s *SqlRetentionPolicyStore) Patch(patch *model.RetentionPolicyWithTeamAndC
+		GroupBy("RetentionPolicies.Id").
+		OrderBy("RetentionPolicies.DisplayName, RetentionPolicies.Id").
+		Limit(uint64(limit)).
+		Offset(uint64(offset))
+
 	if id != "" {
-		whereIdEqualsPolicyId = "WHERE RetentionPolicies.Id = :PolicyId"
-		props["PolicyId"] = id
-	}
-	query = `
-	SELECT RetentionPolicies.Id,
-	       RetentionPolicies.DisplayName,
-	       RetentionPolicies.PostDuration,
-	       A.Count AS ChannelCount,
-	       B.Count AS TeamCount
-	FROM RetentionPolicies
-	INNER JOIN (
-		SELECT RetentionPolicies.Id,
-		       COUNT(RetentionPoliciesChannels.ChannelId) AS Count
-		FROM RetentionPolicies
-		LEFT JOIN RetentionPoliciesChannels ON RetentionPolicies.Id = RetentionPoliciesChannels.PolicyId
-		` + whereIdEqualsPolicyId + `
-		GROUP BY RetentionPolicies.Id
-		ORDER BY RetentionPolicies.DisplayName, RetentionPolicies.Id
-		LIMIT :Limit
-		OFFSET :Offset
-	) AS A ON RetentionPolicies.Id = A.Id
-	INNER JOIN (
-		SELECT RetentionPolicies.Id,
-		       COUNT(RetentionPoliciesTeams.TeamId) AS Count
-		FROM RetentionPolicies
-		LEFT JOIN RetentionPoliciesTeams ON RetentionPolicies.Id = RetentionPoliciesTeams.PolicyId
-		` + whereIdEqualsPolicyId + `
-		GROUP BY RetentionPolicies.Id
-		ORDER BY RetentionPolicies.DisplayName, RetentionPolicies.Id
-		LIMIT :Limit
-		OFFSET :Offset
-	) AS B ON RetentionPolicies.Id = B.Id
-	ORDER BY RetentionPolicies.DisplayName, RetentionPolicies.Id`
-	return
+		rpcSubQuery = rpcSubQuery.Where(sq.Eq{"RetentionPolicies.Id": id})
+	}
+
+	rpcSubQueryString, args, err := rpcSubQuery.ToSql()
+	if err != nil {
+		return "", nil, errors.Wrap(err, "retention_policies_tosql")
+	}
+
+	rptSubQuery := s.getQueryBuilder().
+		Select("RetentionPolicies.Id, COUNT(RetentionPoliciesTeams.TeamId) AS Count").
+		From("RetentionPolicies").
+		LeftJoin("RetentionPoliciesTeams ON RetentionPolicies.Id = RetentionPoliciesTeams.PolicyId").
+		GroupBy("RetentionPolicies.Id").
+		OrderBy("RetentionPolicies.DisplayName, RetentionPolicies.Id").
+		Limit(uint64(limit)).
+		Offset(uint64(offset))
+
+	if id != "" {
+		rptSubQuery = rptSubQuery.Where(sq.Eq{"RetentionPolicies.Id": id})
+	}
+
+	rptSubQueryString, _, err := rptSubQuery.ToSql()
+	if err != nil {
+		return "", nil, errors.Wrap(err, "retention_policies_tosql")
+	}
+
+	query := s.getQueryBuilder().
+		Select(`
+		  RetentionPolicies.Id as "Id",
+			RetentionPolicies.DisplayName,
+			RetentionPolicies.PostDuration,
+		  A.Count AS ChannelCount,
+			B.Count AS TeamCount
+	  `).
+		From("RetentionPolicies").
+		InnerJoin(`(` + rpcSubQueryString + `) AS A ON RetentionPolicies.Id = A.Id`).
+		InnerJoin(`(` + rptSubQueryString + `) AS B ON RetentionPolicies.Id = B.Id`).
+		OrderBy("RetentionPolicies.DisplayName, RetentionPolicies.Id")
+
+	queryString, _, err := query.ToSql()
+	if err != nil {
+		return "", nil, errors.Wrap(err, "retention_policies_tosql")
+	}
+
+	if s.DriverName() == model.DatabaseDriverMysql {
+		args = append(args, args...)
+	}
```

Comment:
Is this because postgres has positional params and mysql is not?

Reply:
Yes, since mysql does not support positional params. I had to put this check.

Reply:
Ok, let's add a comment here to clarify that.